pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -552,7 +552,10 @@ func getLocalIP() ([]v1.NodeAddress, error) {
 			klog.Warningf("Failed to extract addresses for NodeAddresses - %v", err)
 		} else {
 			for _, addr := range localAddrs {
-				if ipnet, ok := addr.(*net.IPNet); ok && !ipnet.IP.IsLoopback() {
```

Comment:
any idea why `ipnet.IP.IsLoopback()` is not behaving correctly here?

Reply:
Because `IsLoopback` is only looking at the ip address, it can only reject `127.0.0.1` and `::1`. Our machines have two ip addresses associated with the loopback interface, so just looking at the ip address isn't enough.